File: 804-unique-morse-code-words/804-unique-morse-code-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function (words) {
  const morseCodes = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."];

  const wordsParsed = words.map(word => {
    return word.split("").map((c) => morseCodes[getCode(c)]).join("");
  });

  return [...new Set(wordsParsed)].length
};

const getCode = (char = '') => {
  return char.charCodeAt(0) - 97
}

File: 804-unique-morse-code-words/804-unique-morse-code-words.go
func uniqueMorseRepresentations(words []string) int {
	morseCodes := []string{".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."}
	result := make(map[string]bool)

	for _, word := range words {
		wordParsed := ""
		for _, c := range word {
			wordParsed += morseCodes[c-97]
		}

		result[wordParsed] = true
	}

	return len(result)
}